File: solutions/elixir/hamming/2/point-mutations.exs
defmodule DNA do
  def hamming_distance(strand, descendant_strand) do
    zip(strand, descendant_strand) |> Enum.count(fn({x1, x2}) -> x1 != x2 end)
  end

  defp zip(strand, descendant_strand) do
    List.zip([strand, descendant_strand])
  end
end

File: solutions/javascript/point-mutations/1/point-mutations.js
"use strict";
var DNA = (function() {
  function DNA(strand) {
    this.hammingDistance = function(descendantStrand) {
      var difference = 0
      if (strand.length > descendantStrand.length) {
        strand = strand.substring(0, descendantStrand.length);
      }
      strand.split('').forEach(function(nucleotide, idx) {
        if (nucleotide !== descendantStrand.charAt(idx)) {
          difference += 1;
        }
      });
      return difference;
    }
  }

  return DNA;
})();

module.exports = DNA;